tests/lib/cli-engine/cli-engine.js

```diff
@@ -3121,18 +3169,29 @@ describe("CLIEngine", () => {
                 `
             };
 
-            it("should lint only JavaScript blocks if '--ext' was not given.", () => {
-                CLIEngine = defineCLIEngineWithInMemoryFileSystem({
-                    cwd: () => root,
+            /* eslint-disable-next-line func-style*/
+            let cleanup = () => {};
+
+            afterEach(async () => {
+                await cleanup();
+                cleanup = () => {};
+            });
+
+            it("should lint only JavaScript blocks if '--ext' was not given.", async () => {
+                const teardown = createCustomTeardown({
+                    cwd: root,
                     files: {
                         ...commonFiles,
-                        ".eslintrc.json": JSON.stringify({
+                        ".eslintrc.json": {
                             plugins: ["markdown", "html"],
                             rules: { semi: "error" }
-                        })
+                        }
                     }
-                }).CLIEngine;
-                engine = new CLIEngine({ cwd: root });
+                });
+
+                await teardown.prepare();
+                cleanup = teardown.cleanup;
```

Comment:
Would it be good to switch the order:

```js
cleanup = teardown.cleanup;
await teardown.prepare();
```

so the cleanup can be done even if prepare maybe partially failed, but still created some files.

Reply:
Done.